By default it passes os.name there, so key is empty and the preferred scheme is ignored.
Pass 'prefix' as the key when it's empty to get the preferred scheme.

diff --git a/setuptools/_distutils/command/install.py b/setuptools/_distutils/command/install.py
index dc17e56a8..9e673d5a3 100644
--- a/setuptools/_distutils/command/install.py
+++ b/setuptools/_distutils/command/install.py
@@ -142,7 +142,7 @@ def _remove_set(ob, attrs):
 def _resolve_scheme(name):
     os_name, sep, key = name.partition('_')
     try:
-        resolved = sysconfig.get_preferred_scheme(key)
+        resolved = sysconfig.get_preferred_scheme(key or 'prefix')
     except Exception:
         resolved = fw.scheme(name)
     return resolved
